### DISCUSSION TITLE
Automatic unlock of locked out user accounts

### DISCUSSION THREAD

**Discussion by ashep** [zitadel/zitadel#8358] [Q&A] [answered]
Hello!

There is a Zitadel functionality that allows the locking of a user account automatically after several unsuccessful authentication attempts.

The question is whether Zitadel allows the unlocking of such accounts automatically after a certain period of time.

If not, do you have any plans to implement this in the near future?

Thank you.

**Comment by fforootd** [comment]
ATM there is no "auto unlock" since we encourage our users to build their own process for that.

On what basis would your needs be to "auto unlock" a user? Timebased?

**Reply by ashep** [reply]
Yep, I need to automatically unlock a locked-out account after some time.

**Reply by fforootd** [reply, marked as answer]
Ok I see, I think the best option is to open a feature request here https://github.com/zitadel/zitadel/issues/new?assignees=&labels=enhancement&projects=&template=proposal.yaml

Until then you can query the zitadel api and unlock users after the set time through the api as well 😁 

The event api could be helpfull for this I guess.

**Reply by ashep** [reply]
Thank you for the help, @fforootd!
